core/src/main/codegen/PolyAlgParser.jj

```diff
@@ -0,0 +1,697 @@
+import org.polypheny.db.algebra.polyalg.parser.nodes.PolyAlgLiteral.LiteralType;
+import org.polypheny.db.algebra.polyalg.parser.nodes.PolyAlgAliasedArgument;
+import org.polypheny.db.algebra.polyalg.parser.nodes.PolyAlgNamedArgument;
+import org.polypheny.db.algebra.polyalg.parser.nodes.PolyAlgNode;
+import org.polypheny.db.algebra.polyalg.parser.nodes.PolyAlgDataType;
+import org.polypheny.db.algebra.polyalg.parser.nodes.PolyAlgOperator;
+import org.polypheny.db.algebra.polyalg.parser.nodes.PolyAlgNodeList;
+import org.polypheny.db.algebra.polyalg.parser.nodes.PolyAlgExpression;
+import org.polypheny.db.algebra.polyalg.parser.nodes.PolyAlgExpressionExtension;
+import org.polypheny.db.algebra.polyalg.parser.nodes.PolyAlgExpressionExtension.ExtensionType;
+import org.polypheny.db.algebra.core.CorrelationId;
+import org.polypheny.db.rex.RexLocalRef;
+import org.polypheny.db.type.entity.PolyString;
+import org.polypheny.db.type.entity.numerical.PolyInteger;
+import org.polypheny.db.type.entity.numerical.PolyDouble;
+import org.polypheny.db.type.entity.PolyValue;
+import org.polypheny.db.type.entity.PolyList;
+import org.polypheny.db.type.entity.document.PolyDocument;
+import org.polypheny.db.type.entity.graph.PolyDictionary;
+import org.polypheny.db.type.entity.graph.PolyNode;
+import org.polypheny.db.type.entity.graph.PolyEdge;
+import org.polypheny.db.type.entity.graph.PolyEdge.EdgeDirection;
+import org.polypheny.db.type.entity.graph.PolyPath;
+import org.polypheny.db.algebra.polyalg.PolyAlgUtils;
+import org.polypheny.db.util.Quadruple;
+import java.util.Arrays;
+import java.util.List;
+import java.util.ArrayList;
+
+/**
+ * Parses Poly Algebra.
+ */
+public class PolyAlgParserImpl extends PolyAlgAbstractParserImpl {
+
+    /**
+     * ParserFactory implementation for creating parser.
+     */
+    public static final ParserFactory FACTORY = new ParserFactory() {
+        public PolyAlgAbstractParserImpl getParser(Reader reader) {
+            final PolyAlgParserImpl parser = new PolyAlgParserImpl(reader);
+          return parser;
+        }
+    };
+
+    /** Main entry point. */
+    public static void parse(String str) {
+        PolyAlgParserImpl parser = new PolyAlgParserImpl(new StringReader(str));
+
+        try {
+          parser.Input();
+          System.out.println("Successfully parsed input!");
+        } catch (Exception e) {
+          System.out.println("Could not parse input correctly:");
+          System.out.println(e.getMessage());
+          e.printStackTrace();
+        }
+    }
+
+    public PolyAlgParseException normalizeException(Throwable ex) {
+        try {
+            return convertException(ex);
+        } catch (ParseException e) {
+            throw new AssertionError(e);
+        }
+    }
+
+    PolyAlgParseException convertException(Throwable ex) throws ParseException {
+        if (ex instanceof PolyAlgParseException) {
+            return (PolyAlgParseException) ex;
```

Comment:
Pattern variable

Reply:
JavaCC sadly struggles with the `instanceof` Pattern variables